File: app/src/main/java/com/chooloo/www/callmanager/ui/contact/ContactPresenter.java
package com.chooloo.www.callmanager.ui.contact;

import com.chooloo.www.callmanager.ui.base.BasePresenter;

public class ContactPresenter<V extends ContactMvpView> extends BasePresenter<V> implements ContactMvpPresenter<V> {
    @Override
    public void onActionCall() {
        mvpView.callContact();
    }

    @Override
    public void onActionSms() {
        mvpView.smsContact();
    }

    @Override
    public void onActionEdit() {
        mvpView.editContact();
    }

    @Override
    public void onActionInfo() {
        mvpView.openContact();
    }

    @Override
    public void onActionDelete() {
        mvpView.deleteContact();
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/contact/ContactPresenter.kt
package com.chooloo.www.callmanager.ui.contact

import com.chooloo.www.callmanager.ui.base.BasePresenter

class ContactPresenter<V : ContactMvpView> : BasePresenter<V>(), ContactMvpPresenter<V> {

    override fun onActionCall() {
        mvpView?.call()
    }

    override fun onActionSms() {
        mvpView?.sms()
    }

    override fun onActionEdit() {
        mvpView?.edit()
    }

    override fun onActionInfo() {
        mvpView?.open()
    }

    override fun onActionDelete() {
        mvpView?.delete()
    }
}